nntrainer/src/tensor.cpp

```diff
@@ -847,15 +841,29 @@ float *Tensor::getAddress(unsigned int i) {
   return &getData()[i];
 }
 
-Tensor &Tensor::copy(Tensor &from) {
+const float *Tensor::getAddress(unsigned int i) const {
+  if (i > this->dim.getDataLen()) {
+    ml_loge("Error: Index out of bounds");
+    return nullptr;
+  }
+
+  return &getData()[i];
+}
+
+void Tensor::copy(const Tensor &from) {
   // todo: enable copy to non-contiguous tensor
   if (!is_contiguous) {
     throw std::runtime_error("Cannot copy non-contiguous tensor");
   }
 
-  *this = from;
+  Tensor t = Tensor(from.getDim(), from.getData());
+  swap(t, *this);
+}
 
-  return *this;
+Tensor Tensor::clone() const {
+  Tensor t;
+  t.copy(*this);
+  return t;
```

Comment:
return `std::move(t)`?

Reply:
I guess compiler optimization takes place here. [#ref](https://developers.redhat.com/blog/2019/04/12/understanding-when-not-to-stdmove-in-c/)